### PR TITLE
Fix #13 - Make docstring optional for deftest

### LINKED ISSUE #13: deftest without docstring silently drops first step
The following test:
```clojure
(deftest my-test
  #::step{:name 'step-1}
  #::step{:name 'step-2})
```
will only execute step 2 because it thinks step 1 is the docstring.

I'd prefer docstrings to be optional in `deftest`s, or for `deftest` to throw an error if the second argument isn't a string.

### PR DESCRIPTION
**Pull request by trevorbernard**

### CODE REVIEW

**Comment by trevorbernard** [comment]
Right, I'll just walk the keys in the first position to test if they being to the test namespace.

**Comment by trevorbernard** [comment]
@jstokes Update the PR to test if the value in the first position is either a docstring or a configuration map.

File: src/greenlight/test.clj [hunk 1]
```diff
@@ -47,20 +47,38 @@
              :kind? vector?
              :min-count 1))
 
+(defn- contains-ns? [coll ns]
+  (not (empty? (reduce-kv (fn [m k v]
+                            (if (= ns (namespace k))
+                              (assoc m k v)
+                              m)) {} coll))))
+
+(defn- attr-map? [o]
```
Comment (jstokes) [inline src/greenlight/test.clj]: Can you put the argument vector on the next line to conform to existing styling?

File: src/greenlight/test.clj [hunk 2]
```diff
@@ -47,20 +47,38 @@
...
+                (rest body)
+                body)
+        base (cond-> {}
+               docstring (assoc ::description docstring)
+               attr-map (merge attr-map))]
+    `(defn ~(vary-meta test-name assoc ::test true)
        []
        (assoc ~base
-              ::title ~(str test-sym)
+              ::title ~(str test-name)
```
[the remark this answers is not in the document]
Comment (trevorbernard) [reply src/greenlight/test.clj]: Ok, I'll revert the name

File: src/greenlight/test.clj (hunk 1 above)
Comment (trevorbernard) [reply src/greenlight/test.clj]: Aye, I will make the change

File: src/greenlight/test.clj [hunk 3]
```diff
@@ -47,20 +47,38 @@
              :kind? vector?
              :min-count 1))
 
+(defn- contains-ns? [coll ns]
+  (not (empty? (reduce-kv (fn [m k v]
```
[the remark this answers is not in the document]
Comment (trevorbernard) [reply src/greenlight/test.clj]: No real reason, I was waiting for a connecting flight and didn't put much thought into it. This is a much better solution.

**Comment by trevorbernard** [comment]
@jstokes submitted a new patch with your suggested changes

**Pull request merged**